src/ivc/public_params.rs

```diff
@@ -185,8 +141,35 @@ where
         }
     }
 
+    /// This method calculate digest of [`PublicParams`], but ignore [`CircuitPublicParams::ck`]
+    /// from both step circuits params
     pub fn digest<C: CurveAffine>(&self) -> Result<C, crate::ivc::Error> {
-        digest::DefaultHasher::digest_to_curve(self).map_err(crate::ivc::Error::WhileHash)
+        #[derive(serde::Serialize)]
+        #[serde(bound(serialize = ""))]
+        struct Wrapper<'l, C1, C2, RP1, RP2>
+        where
+            C1: CurveAffine<Base = <C2 as PrimeCurveAffine>::Scalar> + Serialize,
+            C2: CurveAffine<Base = <C1 as PrimeCurveAffine>::Scalar> + Serialize,
+
+            C1::Base: PrimeFieldBits + FromUniformBytes<64> + Serialize,
+            C2::Base: PrimeFieldBits + FromUniformBytes<64> + Serialize,
+
+            RP1: ROPair<C1::Base>,
+            RP2: ROPair<C2::Base>,
+        {
+            primary_k_table_size: u32,
+            secondary_k_table_size: u32,
+            primary_params: &'l SynthesizeStepParams<C2, RP2::OnCircuit>,
+            secondary_params: &'l SynthesizeStepParams<C1, RP1::OnCircuit>,
+        }
+
+        digest::DefaultHasher::digest_to_curve(&Wrapper::<'_, C1, C2, RP1, RP2> {
+            primary_k_table_size: self.primary.k_table_size,
```

Comment:
Ignore the commitment key is okay.  It seems that `plonk_structure` is also ignored.  We need keep it (and carefully skip some of its fields later).

Reply:
It didn't appear in the final public params at all. We have discussed that for each step a new `TableData` is created (and our API doesn't allow it any other way), so there is simply nothing to create a plonk structure from. 

And plonk structure formed after each step participates in hashes as it is

Reply:
Although each step we synthesize it with different data, but the fixed columns and plonk_structure we are folding are the same. We have to make sure when folding two instances, they share the same plonk_structure e.g. same custom gates.
There is possibility that the prover folded two instances with correct data (fixed+advice column values) but they have different gates or different metadata. We can avoid this kind of potential risk issue by hash metadata defined in `plonk_structure`.